### ISSUE TITLE
OOM while saving the finetune checkpoints

### ISSUE THREAD

**Issue by lzz-dev** [ByteDance-Seed/Bagel#116]
I am doing the t2i finetune following the suggested arguments settings with 8xH800. It can successfully train. When saving the checkpoint, it encounters with OOM sometimes. I was wondering if there is some methods to reduce the VRAM usage, I checked the batch_size which is already set as 1.
Regards.

**Comment by Andy1621** [comment]
Hi! The OOM is caused by saving with FULL_STATE_DICT in fsdp_utils.py

```python
with FSDP.state_dict_type(
    ema_model,
    StateDictType.FULL_STATE_DICT,
    FullStateDictConfig(rank0_only=True, offload_to_cpu=True),
):
```

You can change it to save and load different sharding.

**Comment by lzz-dev** [comment]
> Hi! The OOM is caused by saving with FULL_STATE_DICT in fsdp_utils.py
> 
> with FSDP.state_dict_type(
>     ema_model,
>     StateDictType.FULL_STATE_DICT,
>     FullStateDictConfig(rank0_only=True, offload_to_cpu=True),
> ):
> You can change it to save and load different sharding.

Thanks for your reply.

I ‘ve tried to alter the save_file into FSDP._save_state_to_dict with ShardedStateDictConfig. it encounters some other problems. (See the following detailed modifications). How do you achieve save shadings?

with FSDP.state_dict_type( model, StateDictType.SHARDED_STATE_DICT, ShardedStateDictConfig(offload_to_cpu=True), ): model_state_dict = model.state_dict() 
FSDP._save_to_state_dict(model_state_dict, os.path.join(save_path, "model"))
